website/content/docs/operations/aws-oidc-provider.mdx

```diff
@@ -0,0 +1,361 @@
+To use Nomad as an identity provider, you need to have a trusted SSL certificate for the
+domain used by the cluster. The following example uses AWS Certificate Manager (ACM).
+
+### Create a hosted zone
+
+Use the [`aws_route53_zone` resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_zone).
+
+```hcl
+variable domain_name {
+    type = "string"
+    default = "<DOMAIN_NAME>"
+}
+
+resource "aws_route53_zone" "example" {
+    name = var.domain_name
+}
+```
+
+This configuration requires the following information:
+- [`<DOMAIN_NAME>`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_zone#name):
+The domain name of your Nomad cluster without the protocol or port. `var.domain_name` is used
+throughout the examples on this page to reference `<DOMAIN_NAME>`.
+
+### Generate SSL certificates
+
+Use the [`aws_acm_certificate`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acm_certificate)
+and [`aws_acm_certificate_validation`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acm_certificate_validation) resources to create a request and provide validation for an SSL
+certficiate.
+
+```hcl
+resource "aws_acm_certificate" "example" {
+  domain_name       = var.domain_name
+  validation_method = "DNS"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_acm_certificate_validation" "example" {
+  certificate_arn         = aws_acm_certificate.example.arn
+  validation_record_fqdns = [var.domain_name]
+}
+```
+
+### Create an Application Load Balancer
+
+Use the [`aws_lb` resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb).
+
+```hcl
+resource "aws_lb" "test" {
+  name               = "test-lb-tf"
+  internal           = false
+  load_balancer_type = "application"
+  security_groups    = [<SECURITY_GROUP_ID>]
+  subnets            = [<SUBNET_IDS>]
+
+  enable_deletion_protection = true
+
+  access_logs {
+    bucket  = <S3_BUCKET_ID>
+    prefix  = "test-lb"
+    enabled = true
+  }
+}
+```
+
+This configuration requires the following information:
+- [`<SECURITY_GROUP_ID>`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb#security_groups):
+A list of security groups IDs that you want to apply to the load balancer.
+- [`<SUBNET_IDS>`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb#subnets):
+A list of subnet IDs that you want to attach to the load balancer.
+- [`<S3_BUCKET_ID>`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb#bucket):
+The ID of an S3 bucket to store the load balancer access logs in.
+
+### Create a load balancer listener
+
+Use the [`aws_lb_listener` resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb_listener).
+
+```hcl
+resource "aws_lb_listener" "example" {
+  load_balancer_arn = <LB_ARN>
+  port              = "443"
```

Comment:
The Nomad agents are listening on port 4646, but I don't see how we're mapping port 443 to that?

Reply:
I'm not sure which "original guide" you're referring to here, but I think I also assumed this had been verified end-to-end and not just ported from some other source.

Reply:
Ok, maybe we should pull in @SuyashHashiCorp to take a look at this doc as well then? If I recall correctly that guide may have also been inspired by work @schmichael did.

Reply:
I'm really excited to see workload identity federation (WIF) documentation for another cloud provider, but I'm afraid this content probably needs to be moved to https://github.com/hashicorp/tutorials if it is intended to be complementary to our GCP WIF tutorial: https://developer.hashicorp.com/nomad/tutorials/fed-workload-identity/integration-gcp

Source for that is here https://github.com/hashicorp/tutorials/blob/main/content/tutorials/nomad/integration-gcp.mdx and the corresponding jobspecs/terraform are here: https://github.com/hashicorp-education/learn-nomad-workload-identity-federation/tree/main

[proxy.nomad.hcl](https://github.com/hashicorp-education/learn-nomad-workload-identity-federation/blob/main/proxy.nomad.hcl) is what handles proxying JWKS requests from the cloud vendor to Nomad's HTTP endpoint. You should be able to use that jobspec for any deployment in the cloud or onprem. The flow in GCP is:

```
GCP  ---https (443)---> LB ---http (80)---> Proxy ---task api (unix socket)---> Nomad API
```

...and I assume a similar flow works for any cloud.

---

If there is a desire to keep this content in docs instead of tutorials, I think we should move the GCP tutorial over here and create a subdirectory under `Operations` for both. Ideally we'd add more WIF examples in the future. This all _feels_ more like tutorials to me, but I can understand the desire and simplicity of reference docs. If we have any hard and fast rules about what makes something a tutorial vs a doc, I don't know it, so as long as we're consistent I'm ok with WIF content in either place as long as we're consistent.

Reply:
@schmichael that's totally valid. Education is working on a larger initiative to better organize tutorials and docs and we've got some ongoing plans. @boruszak or I can loop you in for better context but the gist is: there are lots of tutorials/docs that could and should be moved around and edited to better fit in their respective places. 

This doc was being brought over from the support team's knowledge base to make it more visible and we did intend to shuffle it around as part of the larger edu goal when we get there. 😅